apps/web/src/services/iam/components/RoleManagementTabDetail.vue

```diff
@@ -125,25 +123,8 @@ const handleUpdateForm = (value: PageAccessMenuItem) => {
     });
 };
 
-const getRoleDetailData = async (roleId: string) => {
-    state.loading = true;
-    try {
-        state.data = await SpaceConnector.clientV2.identity.role.get<RoleGetParameters, RoleModel>({ role_id: roleId });
-    } catch (e) {
-        ErrorHandler.handleError(e);
-        state.data = {};
-    } finally {
-        state.loading = false;
-    }
-};
-
-watch(() => state.selectedRole.role_id, async (roleId) => {
-    const selectedRoleId = rolePageStore.selectedRoles.length === 0 && rolePageState.selectedIndices.length !== 0
-        ? rolePageState.roles[rolePageState.selectedIndices[0]].role_id
-        : roleId;
-
-    await getRoleDetailData(selectedRoleId);
-    state.pageAccessDataList = getPageAccessMenuListByRoleType(state.data.role_type, state.menuList);
+watch(roleData, async () => {
+    state.pageAccessDataList = getPageAccessMenuListByRoleType(roleData.value?.role_type ?? ROLE_TYPE.DOMAIN_ADMIN, state.menuList);
```

Comment:
Suggestion: How about using `computed` to assigning computed value (`state.pageAccessDataList`) instead of `watch`!